### DISCUSSION TITLE
Local Deployment on MacOS

### DISCUSSION THREAD

**Discussion by DanielaCasilli** [alibaba/GraphScope#611] [Q&A] [answered]
Hi, I am trying to deploy graphscope locally on my Mac however when running the below command, I get a build error

```
./scripts/deploy_local.sh build_and_deploy
export GRAPHSCOPE_HOME=/usr/local
```

Error:
```
package:   /Users/danielacasilli/Docs/Masters/Dissertation/GraphScope/interactive_engine/src/executor/Pegasus/Cargo.toml
workspace: /Users/danielacasilli/Docs/Masters/Dissertation/GraphScope/interactive_engine/src/executor/Cargo.toml
   Compiling serde_json v1.0.57
   Compiling ipnetwork v0.17.0
   Compiling indexmap v1.7.0
   Compiling bincode v1.3.1
   Compiling serde_yaml v0.8.13
   Compiling serde-value v0.5.2
   Compiling serde_yaml v0.7.5
   Compiling toml v0.5.8
   Compiling bstr v0.2.15
   Compiling log v0.4.11
   Compiling which v4.0.2
   Compiling failure v0.1.8
   Compiling prost-types v0.7.0
   Compiling futures-executor v0.3.15
   Compiling structopt v0.3.22
   Compiling grpcio-sys v0.4.7
   Compiling maxgraph-runtime v0.1.0 (/Users/danielacasilli/Docs/Masters/Dissertation/GraphScope/interactive_engine/src/executor/runtime)
error: failed to run custom build command for `maxgraph-runtime v0.1.0 (/Users/danielacasilli/Docs/Masters/Dissertation/GraphScope/interactive_engine/src/executor/runtime)`

Caused by:
  process didn't exit successfully: `/Users/danielacasilli/Docs/Masters/Dissertation/GraphScope/interactive_engine/src/executor/target/debug/build/maxgraph-runtime-0a3669e337b1630c/build-script-build` (exit code: 101)
  --- stdout
  running: "cmake" "/Users/danielacasilli/Docs/Masters/Dissertation/GraphScope/interactive_engine/src/executor/runtime/native" "-DCMAKE_INSTALL_PREFIX=/Users/danielacasilli/Docs/Masters/Dissertation/GraphScope/interactive_engine/src/executor/target/debug/build/maxgraph-runtime-ba1345a973628656/out" "-DCMAKE_C_FLAGS= -ffunction-sections -fdata-sections -fPIC --target=x86_64-apple-darwin" "-DCMAKE_C_COMPILER=/usr/local/opt/llvm@9/bin/clang" "-DCMAKE_CXX_FLAGS= -ffunction-sections -fdata-sections -fPIC --target=x86_64-apple-darwin" "-DCMAKE_CXX_COMPILER=/usr/local/opt/llvm@9/bin/clang++" "-DCMAKE_ASM_FLAGS= -ffunction-sections -fdata-sections -fPIC --target=x86_64-apple-darwin" "-DCMAKE_ASM_COMPILER=/usr/local/opt/llvm@9/bin/clang" "-DCMAKE_BUILD_TYPE=Debug"
  -- Configuring done
  -- Build files have been written to: /Users/danielacasilli/Docs/Masters/Dissertation/GraphScope/interactive_engine/src/executor/target/debug/build/maxgraph-runtime-ba1345a973628656/out/build

  --- stderr
  CMake Error in CMakeLists.txt:
    No known features for CXX compiler

    "Clang"

    version 9.0.1.


  thread 'main' panicked at '
  command did not execute successfully, got: exit code: 1

  build script failed, must exit now', /Users/danielacasilli/.cargo/registry/src/github.com-1ecc6299db9ec823/cmake-0.1.44/src/lib.rs:885:5
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...
error: build failed
make: *** [gie] Error 101
```

Any help would be appreciated as I have tried multiple ways to try get Graphscope to work ...

**Comment by acezen** [comment]
hi, @DanielaCasilli, can you paste your cmake version with `cmake --version`

**Reply by DanielaCasilli** [reply]
cmake version 3.10.0

**Comment by acezen** [comment]
Adding 
```
if(POLICY CMP0025)
  cmake_policy(SET CMP0025 NEW)
endif()
```
after https://github.com/alibaba/GraphScope/blob/ce261cb816e326695d9e92258e1a64d624500828/interactive_engine/src/executor/runtime/native/CMakeLists.txt#L1 make a difference for you?

**Reply by DanielaCasilli** [reply]
Hi @acezen, I have tried implementing your suggestion, but unfortunately no improvement. I did however notice the following in the logs, if that indicates anything:

> cd /Users/danielacasilli/Docs/Masters/Dissertation/GraphScope//analytical_engine/build && \
> 	cmake -DCMAKE_INSTALL_PREFIX=/usr/local -DNETWORKX=OFF -DBUILD_TESTS=OFF .. && \
> 	make -j && \
> 	sudo make install
> -- Could NOT find RDKAFKA (missing: RDKAFKA_INCLUDE_DIR RDKAFKA_LIBRARY) 
> -- Boost  found.
> -- Found Boost components:

**Reply by DanielaCasilli** [reply]
I have tried recloning the project and rerunning the deployment commands but am now seeing the below error:

CMake Warning at cmake/protobuf.cmake:48 (message):
    gRPC_PROTOBUF_PROVIDER is "module" but PROTOBUF_ROOT_DIR is wrong
  Call Stack (most recent call first):
    CMakeLists.txt:140 (include)


  CMake Warning at cmake/gflags.cmake:26 (message):
    gRPC_GFLAGS_PROVIDER is "module" but GFLAGS_ROOT_DIR is wrong
  Call Stack (most recent call first):
    CMakeLists.txt:142 (include)


  CMake Warning (dev):
    Policy CMP0042 is not set: MACOSX_RPATH is enabled by default.  Run "cmake
    --help-policy CMP0042" for policy details.  Use the cmake_policy command to
    set the policy and suppress this warning.

    MACOSX_RPATH is not specified for the following targets:

     zlib

  This warning is for project developers.  Use -Wno-dev to suppress it.

  /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: lib/libcares.a(ares_getenv.c.o) has no symbols
  /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: lib/libcares.a(ares_platform.c.o) has no symbols
  /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: lib/libcares.a(ares_strcasecmp.c.o) has no symbols
  /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: lib/libcares.a(ares_writev.c.o) has no symbols
  /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: lib/libcares.a(windows_port.c.o) has no symbols
  /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: lib/libcares.a(ares_getenv.c.o) has no symbols
  /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: lib/libcares.a(ares_platform.c.o) has no symbols
  /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: lib/libcares.a(ares_strcasecmp.c.o) has no symbols
  /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: lib/libcares.a(ares_writev.c.o) has no symbols
  /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: lib/libcares.a(windows_port.c.o) has no symbols
  clang-9: error: unsupported argument '-g' to option 'Wa,'
  make[4]: *** [third_party/boringssl/crypto/chacha/CMakeFiles/chacha.dir/chacha-x86_64.S.o] Error 1
  make[3]: *** [third_party/boringssl/crypto/chacha/CMakeFiles/chacha.dir/all] Error 2
  make[2]: *** [CMakeFiles/grpc.dir/rule] Error 2
  make[1]: *** [grpc] Error 2
  thread 'main' panicked at '
  command did not execute successfully, got: exit code: 2

  build script failed, must exit now', /Users/danielacasilli/.cargo/registry/src/github.com-1ecc6299db9ec823/cmake-0.1.44/src/lib.rs:885:5
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...
error: build failed
make: *** [gie] Error 101

**Comment by wenyuanyu** [comment]
@DanielaCasilli We will make the local deployment as easy as `pip install` in the coming releases. Stay tuned!

**Comment by acezen** [comment, marked as answer]
@DanielaCasilli The standalone pypi package of GraphScope has been released, you can install GraphScope on Mac by `pip install graphscope` now.
